client-app/pages/catalog/catalog.vue

```diff
@@ -213,54 +214,31 @@
 </template>
 
 <script setup lang="ts">
-import { ref, reactive, onMounted, watch, unref, watchEffect, Ref, shallowRef } from "vue";
-import { useBreakpoints, breakpointsTailwind, useUrlSearchParams, whenever } from "@vueuse/core";
+import { computed, ref, shallowRef, watch, onMounted, watchEffect } from "vue";
+import { breakpointsTailwind, debouncedWatch, useBreakpoints, whenever } from "@vueuse/core";
 import {
   Breadcrumbs,
+  IBreadcrumbsItem,
   PageSize,
   ProductCardGrid,
   ProductCardList,
   ProductSkeletonGrid,
   ProductSkeletonList,
-  ViewMode,
-  useProducts,
-  CategoryTree,
-  ProductsSearchParams,
+  toFilterExpression,
   useCategories,
-  IBreadcrumbsItem,
+  useProducts,
+  useProductsSearchParams,
+  ViewMode,
 } from "@/shared/catalog";
-import { VcCard, VcCheckbox, VcPagination, VcButton } from "@/components";
+import { VcButton, VcCard, VcCardSkeleton, VcCheckbox, VcPagination } from "@/components";
 import { AddToCart } from "@/shared/cart";
-import { Listbox, ListboxButton, ListboxOptions, ListboxOption } from "@headlessui/vue";
-import { useRoute } from "vue-router";
-import _ from "lodash";
+import { Listbox, ListboxButton, ListboxOption, ListboxOptions } from "@headlessui/vue";
+import { useRouteQueryParam } from "@core/composables";
 
-const breakpoints = useBreakpoints(breakpointsTailwind);
-const isMobile = breakpoints.smaller("md");
-const isMobileSidebar = breakpoints.smaller("lg");
-const mobileSidebarVisible = ref(false);
-const sidebarElement = shallowRef<HTMLElement | null>(null);
+const props = defineProps({ categoryKey: String });
 
-const route = useRoute();
-const params = useUrlSearchParams("history", { removeFalsyValues: true });
-const {
-  products,
-  total,
-  loading,
-  pages,
-  termFacets,
-  rangeFacets,
-  selectedFacets,
-  filterStringFromSelectedFacets,
-  getFilterValueFromRangeFacet,
-  fetchProducts,
-} = useProducts();
-const { categoryTree, loadCategoriesTree } = useCategories();
-
-const category: Ref<CategoryTree | undefined> = ref(undefined);
-const keyword: Ref<string> = ref(`${params.keyword || ""}`);
-
-const sortOptions = [
+const itemsPerPageList = [16, 32, 48];
```

Comment:
You have this already defined in @core/constants - maybe it's better to reuse it